Address code review comments

keeratsingh opened the pull request:
- Remove COOKIE2_HEADER
- Store cookies in the concurrent hashmap in the factory for persistance.
- Added end to end test for cookie persistence

Commit:
Address code review comments

- Remove COOKIE2_HEADER
- Store cookies in the concurrent hashmap in the factory for persistance.
- Added end to end test for cookie persistence

jduo commented on java/flight/flight-core/src/main/java/org/apache/arrow/flight/client/ClientCookieMiddleware.java:
```diff
@@ -74,25 +75,32 @@ public void onHeadersReceived(CallHeaders incomingHeaders) {
-    incomingHeaders.getAll(SET_COOKIE2_HEADER).forEach(handleSetCookieHeader);
+    final Iterable<String> setCookieHeaders = incomingHeaders.getAll(SET_COOKIE_HEADER);
+    if (setCookieHeaders != null) {
+      for (String cookieHeader : setCookieHeaders) {
```
setCookieHeaders.forEach(handleSetCookieHeader)

keeratsingh replied:
Done, that looks cleaner.

jduo commented on java/flight/flight-core/src/test/java/org/apache/arrow/flight/client/TestCookieHandling.java:
```diff
@@ -29,37 +45,47 @@
+
+  @Before
+  public void setup() throws Exception {
+    allocator = new RootAllocator(Long.MAX_VALUE);
```
You must clean up the allocator, server, and client.

keeratsingh replied:
Done.

jduo commented on java/flight/flight-core/src/test/java/org/apache/arrow/flight/client/TestCookieHandling.java:
```diff
@@ -138,27 +164,75 @@ public void multipleCookiesWithSetCookie() {
-    Assert.assertEquals("firstKey=firstVal", cookieMiddleware.calculateCookieString());
+  public void cookieStaysAfterMultipleRequestsEndToEnd() {
+    client.handshake();
+    Assert.assertEquals("k=v", cookieMiddleware.getValidCookiesAsString());
```
These tests are wrong. You are just checking getValidCookiesAsString() on the same cookieMiddleware instead of new ones the server makes.

keeratsingh replied:
Done, updated the test co check getValidCookiesAsString on newly created middlewares.

Commit:
Fixed test cookieStaysAfterMultipleRequestsEndToEnd

- Fixed test cookieStaysAfterMultipleRequestsEndToEnd.
- Cleanup TestCookieHandling resources.

jduo commented on java/flight/flight-core/src/main/java/org/apache/arrow/flight/client/ClientCookieMiddleware.java:
```diff
@@ -37,29 +37,36 @@
-  private Map<String, HttpCookie> cookies = new HashMap<>();
-
-  public ClientCookieMiddleware() {
+  public ClientCookieMiddleware(Factory factory) {
```
Let's make this package private instead of public.

jduo replied:
And mark @VisibleForTesting

keeratsingh replied:
Done.

jduo commented on java/flight/flight-core/src/main/java/org/apache/arrow/flight/client/ClientCookieMiddleware.java:
```diff
@@ -37,29 +37,36 @@
     @Override
     public ClientCookieMiddleware onCallStarted(CallInfo info) {
-      return new ClientCookieMiddleware();
+      this.clientCookieMiddleware = new ClientCookieMiddleware(this);
```
Don't do this in the actual production code.
In your test code, create a subclass of ClientCookieMiddleware.Factory that overrides onCallStarted() that lets you intercept the returned ClientCookieMiddleware.

keeratsingh replied:
Done, added a new subclass in the test.

jduo commented on java/flight/flight-core/src/test/java/org/apache/arrow/flight/client/TestCookieHandling.java:
```diff
@@ -138,27 +169,75 @@ public void multipleCookiesWithSetCookie() {
+      @Override
+      public SetCookieHeaderInjector onCallStarted(CallInfo info, CallHeaders incomingHeaders,
+                                                   RequestContext context) {
+        if (null != incomingHeaders.get(COOKIE_HEADER)) {
```
recceivedCookieHeader = null != incomingHeaders.get(...)

keeratsingh replied:
Done.

Commit:
Address code review feedback

- Update TestCookieHandling to use the test Factory with the ability to intercept the returned ClientCookieMiddleware.

jduo commented:
LGTM
